### PR TITLE
STITCH-3299 - Add admin sdk method for deleting pending users

### PR DESCRIPTION
**Pull request by cktang88**

### CODE REVIEW

File: test/userRegistrations.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  afterEach(async() => th.cleanup());
+
+  it('removePendingUserByEmail should remove existing user', async() => {
+    await client.register(testEmail, password);
+    let users = await th.app().userRegistrations().listPending();
+    expect(users).toHaveLength(2);
+    await th.app().userRegistrations().removePendingUserByEmail(testEmail);
+    users = await th.app().userRegistrations().listPending();
+    expect(users).toHaveLength(1);
```
Comment (mburdette) [inline test/userRegistrations.test.js]: you should actually check the users here to make sure the correct one was removed, can probably find some other assertion to do it or make a new list or something and do an equality check. 

https://jestjs.io/docs/en/expect

File: test/userRegistrations.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  });
+  it('removePendingUserByID should remove existing user', async() => {
+    await client.register(testEmail, password);
+    let users = await th.app().userRegistrations().listPending();
+    expect(users).toHaveLength(2);
+    await th.app().userRegistrations().removePendingUserByID(users[0]._id);
+    users = await th.app().userRegistrations().listPending();
+    expect(users).toHaveLength(1);
```
Comment (mburdette) [inline test/userRegistrations.test.js]: you should actually check the users here to make sure the correct one was removed

File: test/userRegistrations.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  const testEmail = 'link_user@10gen.com';
+  const password = '123456';
+
+  beforeEach(async() => {
+    const { apiKey, groupId, serverUrl } = extractTestFixtureDataPoints(test);
+    th = await buildClientTestHarness(apiKey, groupId, serverUrl);
+
+    client = th.stitchClient;
+    // console.log(client);
```
Comment (mburdette) [inline test/userRegistrations.test.js]: errant comment

File: test/userRegistrations.test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    await client.logout();
+  });
+
+  afterEach(async() => th.cleanup());
+
+  it('removePendingUserByEmail should remove existing user', async() => {
```
Comment (mburdette) [inline test/userRegistrations.test.js]: `should remove an existing pending user`

File: test/userRegistrations.test.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    await client.register(testEmail, password);
+    let users = await th.app().userRegistrations().listPending();
+    expect(users).toHaveLength(2);
+    await th.app().userRegistrations().removePendingUserByEmail(testEmail);
+    users = await th.app().userRegistrations().listPending();
+    expect(users).toHaveLength(1);
+  });
+  it('removePendingUserByID should remove existing user', async() => {
```
Comment (mburdette) [inline test/userRegistrations.test.js]: `should remove an existing pending user`

File: dist/node/admin.js [hunk 6]
```diff
@@ -501,8 +501,14 @@ var StitchAdminClient = exports.StitchAdminClient = function (_StitchClient) {
                 confirmByEmail: function confirmByEmail(email) {
                   return api._post(appUrl + '/user_registrations/by_email/' + email + '/confirm');
                 },
-                listUnconfirmed: function listUnconfirmed(filter) {
-                  return api._get(appUrl + '/user_registrations/unconfirmed_users', filter);
+                listPending: function listPending(filter) {
```
Comment (mburdette) [inline dist/node/admin.js]: i think you need to rebuild and then push, this still says `listUnconfirmed` from before, which is incorrect

Comment (cktang88) [reply dist/node/admin.js]: the diff seems to say listPending replaces listUnconfirmed in this file now?

Comment (mburdette) [reply dist/node/admin.js]: i guess someone forgot to rebuild

**Pull request merged**